third-party/realdds/src/dds-serialization.cpp

```diff
@@ -242,4 +350,36 @@ void override_endpoint_qos_from_json( eprosima::fastdds::dds::RTPSEndpointQos &
 }
 
 
+void override_participant_qos_from_json( eprosima::fastdds::dds::DomainParticipantQos & qos, nlohmann::json const & j )
+{
+    if( ! j.is_object() )
+        return;
+    rsutils::json::get_ex( j, "participant-id", &qos.wire_protocol().participant_id );
+    rsutils::json::get_ex( j, "lease-duration", &qos.wire_protocol().builtin.discovery_config.leaseDuration );
+
+    rsutils::json::get_ex( j, "use-builtin-transports", &qos.transport().use_builtin_transports );
+    if( auto udp_j = rsutils::json::nested( j, "udp" ) )
+    {
+        for( auto t : qos.transport().user_transports )
+            if( auto udp_t = std::dynamic_pointer_cast< eprosima::fastdds::rtps::UDPTransportDescriptor >( t ) )
```

Comment:
1. What if no udp transport is registered, I think we need to create one and push to `user_transports` see example at end of https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/udp/udp.html
2. Using UDPv4 IPs. Is there also requirement for UDPv6?
3. If using user transports don't we need to make sure builtin transports are disabled?

Reply:
I'm just trying to find the UDP transport, if it exists. We don't create transports here, or at least not yet -- just overriding settings for them.

Reply:
For IPv6... we currently specifically add v4. I guess we can address generic transport configuration in another PR, but it's way more advanced.

Reply:
I guess it works because we specifically create UDPv4 transport in the participant creation. OK for now.